[BUG] Task `chp.extension-SH` fails in eGon2035

Apparently at least one CHP was not allocated..

```
...
  File "/storage/428_reGon/eGon-data/SH1_v2.0.0._pre-testrun_JA/code/src/egon/data/datasets/chp/__init__.py", line 674, in extension_SH
    extension_per_federal_state("SchleswigHolstein", EgonChp)
  File "/storage/428_reGon/eGon-data/SH1_v2.0.0._pre-testrun_JA/code/src/egon/data/datasets/chp/small_chp.py", line 606, in extension_per_federal_state
    not_distributed_capacity_industry = extension_industrial(
  File "/storage/428_reGon/eGon-data/SH1_v2.0.0._pre-testrun_JA/code/src/egon/data/datasets/chp/small_chp.py", line 508, in extension_industrial
    not_distributed_capacity = extension_to_areas(
  File "/storage/428_reGon/eGon-data/SH1_v2.0.0._pre-testrun_JA/code/src/egon/data/datasets/chp/small_chp.py", line 222, in extension_to_areas
    selected_areas.loc[:, "bus_id"] = assign_bus_id(
  File "/storage/428_reGon/eGon-data/SH1_v2.0.0._pre-testrun_JA/code/src/egon/data/datasets/power_plants/__init__.py", line 587, in assign_bus_id
    assert power_plants.bus_id.notnull().all(), f"""Some power plants are
AssertionError: Some power plants are
    not attached to a bus:            demand   osm_id  ... voltage_level bus_id
369  34832.302726  1425843  ...           5.0    NaN
[1 rows x 7 columns]
```

--- a/src/egon/data/datasets/chp/small_chp.py
+++ b/src/egon/data/datasets/chp/small_chp.py
@@ -467,7 +467,8 @@ def extension_industrial(federal_state, additional_capacity, flh_chp, EgonChp):
     industry_areas = db.select_geodataframe(
         f"""
         SELECT
-        SUM(demand) as demand, a.osm_id, ST_Centroid(b.geom) as geom, b.name
+        SUM(demand) as demand, a.osm_id,
+        ST_PointOnSurface(b.geom) as geom, b.name
         FROM
         {sources['industrial_demand_osm']['schema']}.
         {sources['industrial_demand_osm']['table']} a,

--- a/src/egon/data/datasets/chp_etrago.py
+++ b/src/egon/data/datasets/chp_etrago.py
@@ -206,6 +206,10 @@ def insert_scenario(scenario):
             electrical_bus_id, ch4_bus_id, a.carrier, c.bus_id)
         """
     )
+
+    chp_dh.loc[chp_dh[chp_dh.carrier=="gas extended"].index,
+                   "carrier"] = "gas"
+
     # Divide into biomass and gas CHP which are modelled differently
     chp_link_dh = chp_dh[chp_dh.carrier == "gas"].index
     chp_generator_dh = chp_dh[chp_dh.carrier != "gas"].index
@@ -345,6 +349,10 @@ def insert_scenario(scenario):
         GROUP BY (electrical_bus_id, ch4_bus_id, carrier)
         """
     )
+
+    chp_industry.loc[chp_industry[chp_industry.carrier=="gas extended"].index,
+                   "carrier"] = "gas"
+
     chp_link_ind = chp_industry[chp_industry.carrier == "gas"].index
 
     chp_generator_ind = chp_industry[chp_industry.carrier != "gas"].index

--- a/src/egon/data/datasets/heat_supply/individual_heating.py
+++ b/src/egon/data/datasets/heat_supply/individual_heating.py
@@ -656,6 +656,18 @@ def cascade_per_technology(
             {"bus_id": "mv_grid_id", "share": "capacity"}, axis=1, inplace=True
         )
 
+    elif (tech.index == "gas_boiler") & (scenario == "eGon2035"):
+        append_df = pd.DataFrame(
+            data={
+                "capacity": heat_per_mv.remaining_demand.div(
+                    tech.estimated_flh.values[0]
+                ),
+                "carrier": f"residential_rural_{tech.index}",
+                "mv_grid_id": heat_per_mv.index,
+                "scenario": scenario,
+            }
+        )
+
     elif tech.index in ("gas_boiler", "resistive_heater", "solar_thermal"):
         # Select target value for Germany
         target = db.select_dataframe(

